pkg/apis/core/types_shoot.go

```diff
@@ -972,9 +972,9 @@ type Maintenance struct {
 // MaintenanceAutoUpdate contains information about which constraints should be automatically updated.
 type MaintenanceAutoUpdate struct {
 	// KubernetesVersion indicates whether the patch Kubernetes version may be automatically updated (default: true).
-	KubernetesVersion bool
+	KubernetesVersion *bool
 	// MachineImageVersion indicates whether the machine image version may be automatically updated (default: true).
-	MachineImageVersion bool
+	MachineImageVersion *bool
 }
```

Comment:
Can you add a method to the Shoot struct:

```go
func (s *Shoot) IsWorkerless() bool {
   return len(s.Spec.Provider.Workers) == 0
```
This has already been repeated in 3 or more places. Having a single place to determine this is always cleaner and enforces DRY. You can also then write a unit test for it as well.